downstreamadapter/dispatcher/redo_dispatcher.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dispatcher
+
+import (
+	"math/rand"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/downstreamadapter/sink"
+	"github.com/pingcap/ticdc/eventpb"
+	"github.com/pingcap/ticdc/heartbeatpb"
+	"github.com/pingcap/ticdc/pkg/common"
+	commonEvent "github.com/pingcap/ticdc/pkg/common/event"
+	"go.uber.org/zap"
+)
+
+type RedoDispatcher struct {
+	*BasicDispatcher
+	isRemoving atomic.Bool
+}
+
+func NewRedoDispatcher(
+	changefeedID common.ChangeFeedID,
+	id common.DispatcherID,
+	tableSpan *heartbeatpb.TableSpan,
+	redoSink sink.Sink,
+	startTs uint64,
+	statusesChan chan TableSpanStatusWithSeq,
+	blockStatusesChan chan *heartbeatpb.TableSpanBlockStatus,
+	schemaID int64,
+	schemaIDToDispatchers *SchemaIDToDispatchers,
+	timezone string,
+	integrityConfig *eventpb.IntegrityConfig,
+	filterConfig *eventpb.FilterConfig,
+	errCh chan error,
+	bdrMode bool,
+) *RedoDispatcher {
+	basicDispatcher := NewBasicDispatcher(
+		changefeedID,
+		id, tableSpan, redoSink,
+		startTs,
+		statusesChan,
+		blockStatusesChan,
+		schemaID,
+		schemaIDToDispatchers,
+		timezone,
+		integrityConfig,
+		nil,
+		false,
+		filterConfig,
+		0,
+		errCh,
+		bdrMode,
+		TypeDispatcherRedo,
+	)
+	dispatcher := &RedoDispatcher{
+		BasicDispatcher: basicDispatcher,
+	}
+
+	return dispatcher
+}
+
+// HandleEvents can batch handle events about resolvedTs Event and DML Event.
+// While for DDLEvent and SyncPointEvent, they should be handled separately,
+// because they are block events.
+// We ensure we only will receive one event when it's ddl event or sync point event
+// by setting them with different event types in DispatcherEventsHandler.GetType
+// When we handle events, we don't have any previous events still in sink.
+func (rd *RedoDispatcher) HandleEvents(dispatcherEvents []DispatcherEvent, wakeCallback func()) (block bool) {
```

Comment:
The main logic here is similar to the ```handleEvents``` function in event dispatcher, can we just reuse the same logic?